lib/logstasher.rb

```diff
@@ -61,6 +63,9 @@ def log_as_json(payload, as_logstash_event: false)
     end
 
     def logger
+      # Return a /dev/null logger if logstasher is enabled or silencing was enabled.
+      return SILENT_LOGGER if !enabled? || silence_standard_logging?
+
       @logger ||= initialize_logger
```

Comment:
I'm afraid we can't change this without potentially breaking someone, but I really think `initialize_logger` should be named `default_logger` or `backup_logger` or something and ought to be private.

If anyone is actually calling it from outside this class, due to its name they might think calling it actually changes @logger, which it doesn't.

So..., given that, maybe it is a good thing if renaming that and making it private breaks anyone who was trying to call it.

Note also that as part of making this private, it doesn't really need configurable params.